File: test/unit/core/helm/model/TestChartOptionsBuilderTest.java
/*
 * Copyright (C) 2023 Hedera Hashgraph, LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.hedera.fullstack.helm.client.test.model;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.mockito.ArgumentMatchers.anyString;
import static org.mockito.Mockito.times;
import static org.mockito.Mockito.verify;

import com.hedera.fullstack.helm.client.execution.HelmExecutionBuilder;
import com.hedera.fullstack.helm.client.model.test.TestChartOptions;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.Mock;
import org.mockito.junit.jupiter.MockitoExtension;

@ExtendWith(MockitoExtension.class)
class TestChartOptionsBuilderTest {
    @Mock
    private HelmExecutionBuilder builderMock;

    @Test
    @DisplayName("Test TestChartOptionsBuilder")
    void testTestChartOptionsBuilder() {
        TestChartOptions options =
                TestChartOptions.builder().filter("filter").timeout("timeout").build();
        assertNotNull(options);
        assertEquals("timeout", options.timeout());
        assertEquals("filter", options.filter());

        options.apply(builderMock);

        verify(builderMock, times(2)).argument(anyString(), anyString());
    }
}

File: test/unit/core/helm/model/TestChartOptionsBuilder.test.ts
// SPDX-License-Identifier: Apache-2.0

import {expect} from 'chai';
import sinon from 'sinon';
import {describe, it} from 'mocha';
import {type HelmExecutionBuilder} from '../../../../../src/integration/helm/execution/HelmExecutionBuilder.js';
import {TestChartOptionsBuilder} from '../../../../../src/integration/helm/model/test/TestChartOptionsBuilder.js';

describe('TestChartOptionsBuilder Tests', () => {
  it('Test TestChartOptionsBuilder', () => {
    const options = TestChartOptionsBuilder.builder().filter('filter').timeout('timeout').build();

    // Verify all options are set correctly
    expect(options).to.not.be.null;
    expect(options.timeout).to.equal('timeout');
    expect(options.filter).to.equal('filter');

    // Test apply method with mock
    const builderMock = {
      argument: sinon.stub().returnsThis(),
    } as unknown as HelmExecutionBuilder;

    options.apply(builderMock);

    // Verify mock interactions
    expect(builderMock.argument).to.have.been.calledTwice;
    expect(builderMock.argument).to.have.been.calledWith('timeout', 'timeout');
    expect(builderMock.argument).to.have.been.calledWith('filter', 'filter');
  });
});
